src/cmd/services/m3coordinator/downsample/options.go

```diff
@@ -161,6 +161,10 @@ func NewAutoMappingRules(namespaces []m3.ClusterNamespace) ([]AutoMappingRule, e
 			continue
 		}
 
+		if opts.ReadOnly() {
```

Comment:
Maybe we need to add some validation code in case someone tries to have readonly: true and downsample:all true?

Reply:
Why? I think these are two independent properties. 
In fact, `readonly: true` and `downsample.all: true` is the combination I am planning to use.